src/lib/GenericPredictor.cc

```diff
@@ -4,17 +4,18 @@
 
 namespace simeng {
 
-GenericPredictor::GenericPredictor(YAML::Node config)
-    : btbBits_(config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint64_t>()),
-      btb_(1 << btbBits_,
-           {config["Branch-Predictor"]["Fallback-Static-Predictor"]
-                .as<uint16_t>(),
-            0}),
-      satCntBits_(
-          config["Branch-Predictor"]["Saturating-Count-Bits"].as<uint64_t>()),
-      globalHistoryLength_(
-          config["Branch-Predictor"]["Global-History-Length"].as<uint64_t>()),
-      rasSize_(config["Branch-Predictor"]["RAS-entries"].as<uint64_t>()) {
+GenericPredictor::GenericPredictor() {
+  YAML::Node& config = Config::get();
+  btbBits_ = config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint64_t>();
+  btb_ = std::vector<std::pair<uint8_t, uint64_t>>(
```

Comment:
I'm unsure on how to do it another way correctly given we now need to fetch the config file, rather than pass it around the object. The alternative is to still pass the config into the constructor function, but have it default to `Config::get()` but I don't personally like to do this if possible.

Additionally, the original construction seems somewhat broken as there is a mix of types (uint8 and uint16)

Reply:
Would it not just be,
```
btb_(1 << btbBits_, {config["Branch-Predictor"]["Fallback-Static-Predictor"].as<uint16_t>(), 0}),
```
As before?

Reply:
It can't be this, as we must define `btb_` inside the constructor (i.e. after Config file has been retrieved) rather than in the initialiser list